app/tournament/tournament.js

```diff
@@ -0,0 +1,93 @@
+import {Player, validatePlayer} from "./player";
+import {Match, MatchFromPrevRound} from "./match";
+import * as utils from "../common/utils";
+
+export class Tournament {
+    constructor() {
+        this.players = [];
+        this.matches = [];
+    }
+
+    addPlayer(player) {
+        this.players.push(player);
+    }
+
+    addMatch(match) {
+        this.matches.push(match);
+    }
+
+    getNumberOfRounds() {
+        return this.matches[this.matches.length - 1].round;
+    }
+
+    generate() {
+        generateTournamentBracket(this);
+    }
+
+    validatePlayer(player) {
+        var validationResult = validatePlayer(player);
+
+        if (validationResult.success && this.existsPlayer(player)) {
+            validationResult.error.push("Player already enrolled");
+            validationResult.success = false;
+        }
+
+        return validationResult;
+    }
+
+    existsPlayer(player) {
+        return undefined !== this.players.find( function(p) {
+            return p.name.toLowerCase() === this.name.toLowerCase()
+            || p.rank === this.rank;
+        }, player);
```

Comment:
You can use player in the function and avoid this second argument. Thus it will read simpler.